brainscore_language/plugins/linear_predictivity/metric.py

```diff
@@ -0,0 +1,180 @@
+        assert set(assembly.dims) == set(self._expected_dims), \
+            f"Expected {set(self._expected_dims)}, but got {set(assembly.dims)}"
+        return assembly.transpose(*self._expected_dims)
+
+
+class XarrayCorrelation:
+    def __init__(self, correlation, correlation_coord=Defaults.stimulus_coord, neuroid_coord=Defaults.neuroid_coord):
+        self._correlation = correlation
+        self._correlation_coord = correlation_coord
+        self._neuroid_coord = neuroid_coord
+
+    def __call__(self, prediction, target) -> Score:
+        # align
+        prediction = prediction.sortby([self._correlation_coord, self._neuroid_coord])
+        target = target.sortby([self._correlation_coord, self._neuroid_coord])
+        assert np.array(prediction[self._correlation_coord].values == target[self._correlation_coord].values).all()
+        assert np.array(prediction[self._neuroid_coord].values == target[self._neuroid_coord].values).all()
+        # compute correlation per neuroid
+        neuroid_dims = target[self._neuroid_coord].dims
+        assert len(neuroid_dims) == 1
+        correlations = []
+        for i, coord_value in enumerate(target[self._neuroid_coord].values):
+            target_neuroids = target.isel(**{neuroid_dims[0]: i})  # `isel` is about 10x faster than `sel`
+            prediction_neuroids = prediction.isel(**{neuroid_dims[0]: i})
+            r, p = self._correlation(target_neuroids, prediction_neuroids)
+            correlations.append(r)
+        # package
+        result = Score(correlations,
+                       coords={coord: (dims, values)
+                               for coord, dims, values in walk_coords(target) if dims == neuroid_dims},
+                       dims=neuroid_dims)
+        return result
+
+
+class CrossRegressedCorrelation(Metric):
+    def __init__(self, regression, correlation, crossvalidation_kwargs=None, store_regression_weights=False):
+        crossvalidation_defaults = dict(train_size=.9, test_size=None)
+        crossvalidation_kwargs = {**crossvalidation_defaults, **(crossvalidation_kwargs or {})}
+
+        self.cross_validation = CrossValidation(**crossvalidation_kwargs)
+        self.regression = regression
+        self.correlation = correlation
+        self.store_regression_weights = store_regression_weights
+
+    def __call__(self, assembly1, assembly2) -> Score:
+        return self.cross_validation(assembly1, assembly2, apply=self.apply, aggregate=self.aggregate)
+
+    def apply(self, source_train, target_train, source_test, target_test):
+        self.regression.fit(source_train, target_train)
+        prediction = self.regression.predict(source_test)
+        score = self.correlation(prediction, target_test)
+        if self.store_regression_weights:
+            self.attach_regression_weights(score=score, source_test=source_test, target_test=target_test)
+        return score
+
+    def attach_regression_weights(self, score, source_test, target_test):
```

Comment:
awesome!! so this would be for just normal linear regression with no hyper parameters (such as alpha ridge regression values)?

Reply:
it should work for all linear regressions using `intercept + weights`, but might have to be adapted based on the variable names yes